editor/animation_track_editor.cpp

```diff
@@ -4066,6 +4066,29 @@ int AnimationTrackEditor::_confirm_insert(InsertData p_id, int p_last_track, boo
 		undo_redo->add_undo_method(this, "_clear_selection", false);
 		undo_redo->add_undo_method(animation.ptr(), "remove_track", animation->get_track_count());
 		p_last_track++;
+
+		// Hack.
+		NodePath np;
+		animation->add_track(p_id.type);
+		animation->track_set_path(animation->get_track_count() - 1, p_id.path);
+		PropertyInfo h = _find_hint_for_track(animation->get_track_count() - 1, np);
+		animation->remove_track(animation->get_track_count() - 1); //hack
+
+		if (h.usage & PROPERTY_USAGE_KEYING_INCREMENTS) {
+
+			// move timeline by one step
+			float step = animation->get_step();
+			if (step == 0)
+				step = 1;
```

Comment:
The current master branch has new style rules, including enforcing braces. Your PR will need to be rebased and updated to conform to these in order for it to be accepted. For more information, read this documentation article: https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html